grails-datastore-core/src/main/groovy/org/grails/datastore/mapping/model/AutoTimestampUtils.java

```
@@ -0,0 +1,172 @@
+                        case CREATED_DATE_SPRING_ANNOTATION:
+                            return AuditMetadataType.CREATED;
+
+                        case LAST_MODIFIED_DATE_ANNOTATION:
+                        case LAST_MODIFIED_DATE_SPRING_ANNOTATION:
+                            return AuditMetadataType.UPDATED;
+
+                        case CREATED_BY_ANNOTATION:
+                        case CREATED_BY_SPRING_ANNOTATION:
+                            return AuditMetadataType.CREATED_BY;
+
+                        case LAST_MODIFIED_BY_ANNOTATION:
+                        case LAST_MODIFIED_BY_SPRING_ANNOTATION:
+                            return AuditMetadataType.UPDATED_BY;
+
+                        case AUTO_TIMESTAMP_ANNOTATION:
+                            // For @AutoTimestamp, check the EventType value
+                            try {
+                                Object eventTypeValue = annotation.annotationType()
+                                    .getMethod("value")
+                                    .invoke(annotation);
+
+                                if (eventTypeValue != null) {
+                                    String eventTypeName = eventTypeValue.toString();
+                                    if (eventTypeName.equals("UPDATED")) {
+                                        return AuditMetadataType.UPDATED;
+                                    } else {
+                                        return AuditMetadataType.CREATED;
+                                    }
+                                }
+                            } catch (Exception e) {
+                                // If we can't read the value, default to CREATED
+                                return AuditMetadataType.CREATED;
+                            }
+                            break;
+                    }
+                }
+            }
+        } catch (Exception ignored) {
+            // If reflection fails, return NONE
```

Would it be useful to add log.debug for these ignored exceptions?